src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.github.lindenb.jbwa.jni.AlnRgn;
+import com.github.lindenb.jbwa.jni.BwaIndex;
+import com.github.lindenb.jbwa.jni.BwaMem;
+import com.github.lindenb.jbwa.jni.ShortRead;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.TextCigarCodec;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.bwa.BWANativeLibrary;
+import scala.Tuple2;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ContigAligner implements Closeable {
+
+    final BwaIndex index;
+    final BwaMem bwaMem;
+
+    public ContigAligner(final File referenceFile) {
+        BWANativeLibrary.load();
+        try {
+            index = new BwaIndex(referenceFile);
+        } catch (IOException e) {
```

Comment:
Local superstition seems to dictate that we noise up the code with `final` everywhere that it can legally go.  If you want to change this to `final IOException e` you should also fix line 82 of AlignContigsAndCallBreakpointsSpark.